universal-application-tool-0.0.1/app/services/program/test/FakeProgramService.java

```diff
@@ -0,0 +1,163 @@
+package services.program.test;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import services.program.*;
+import services.question.QuestionDefinition;
+
+/**
+ * A fake implementation of {@link ProgramService} that uses an in-memory map instead of a database.
+ */
+public class FakeProgramService implements ProgramService {
+
+  private final Map<Long, ProgramDefinition> programs = new HashMap<>();
+  private long nextId = 1L;
+
+  @Override
+  public ImmutableList<ProgramDefinition> listProgramDefinitions() {
+    return ImmutableList.copyOf(this.programs.values());
+  }
+
+  @Override
+  public CompletionStage<ImmutableList<ProgramDefinition>> listProgramDefinitionsAsync() {
+    return CompletableFuture.completedFuture(listProgramDefinitions());
+  }
+
+  @Override
+  public Optional<ProgramDefinition> getProgramDefinition(long id) {
+    return Optional.ofNullable(this.programs.get(id));
+  }
+
+  @Override
+  public CompletionStage<Optional<ProgramDefinition>> getProgramDefinitionAsync(long id) {
+    return CompletableFuture.completedFuture(getProgramDefinition(id));
+  }
+
+  @Override
+  public ProgramDefinition createProgramDefinition(String name, String description) {
+    ProgramDefinition programDefinition =
+        ProgramDefinition.builder().setId(nextId).setName(name).setDescription(description).build();
+    this.programs.put(nextId, programDefinition);
+    this.nextId += 1;
+    return programDefinition;
+  }
+
+  @Override
+  public ProgramDefinition addBlockToProgram(
+      long programId, String blockName, String blockDescription) throws ProgramNotFoundException {
+    ProgramDefinition programDefinition = getProgramOrThrow(programId);
+    BlockDefinition blockDefinition =
+        BlockDefinition.builder()
+            .setId(getNextBlockId(programDefinition))
+            .setName(blockName)
+            .setDescription(blockDescription)
+            .build();
+    programDefinition = programDefinition.toBuilder().addBlockDefinition(blockDefinition).build();
+    this.programs.put(programId, programDefinition);
+    return programDefinition;
```

Comment:
nit: a newline before the return statement makes these easier for me to read

Reply:
Done! Added newlines before all the return statements